extensions/amp-bodymovin-player/0.1/amp-bodymovin-player.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const TAG = 'amp-bodymovin-player';
+
+import {Services} from '../../../src/services';
+import {batchFetchJsonFor} from '../../../src/batched-json';
+import {getIframe, preloadBootstrap} from '../../../src/3p-frame';
+import {isLayoutSizeDefined} from '../../../src/layout';
+import {listenFor} from '../../../src/iframe-helper';
+import {removeElement} from '../../../src/dom';
+import {user} from '../../../src/log';
+
+export class AmpBodymovinPlayer extends AMP.BaseElement {
+
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private @const */
+    this.ampdoc_ = Services.ampdoc(this.element);
+    /** @private {?HTMLIFrameElement} */
+    this.iframe_ = null;
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return isLayoutSizeDefined(layout);
+  }
+
+  /**
+   * @param {boolean=} opt_onLayout
+   * @override
+   */
+  preconnectCallback(opt_onLayout) {
+    preloadBootstrap(this.win, this.preconnect);
+    this.preconnect.url('https://cdnjs.cloudflare.com/ajax/libs/bodymovin/4.13.0/bodymovinjs', opt_onLayout);
+  }
+
+  /** @override */
+  layoutCallback() {
+    this.element.setAttribute('data-loop', this.element.getAttribute('loop'));
+    this.element.setAttribute('data-autoplay',
+        this.element.hasAttribute('no-autoplay') ? false : true);
+    console.log(this.element.getAttribute('data-autoplay'))
+
+    user().assert(
+        !this.element.hasAttribute('src'),
+        'The src attribute must be specified for <amp-bodymovin-player>');
+    if (this.element.hasAttribute('src')) {
```

Comment:
no need for `if` anymore, asserts throw sync, so won't get here in case of errors.